tools/run_tests/xds_k8s_test_driver/framework/infrastructure/gcp/network_services.py

```diff
@@ -22,9 +23,134 @@
 
 logger = logging.getLogger(__name__)
 
+_ComputeV1 = gcp.compute.ComputeV1
+GcpResource = _ComputeV1.GcpResource
+
+@dataclasses.dataclass(frozen=True)
+class Router:
+
+    name: str
+    url: str
+    type: str
+    network: Optional[str]
+    routes: Optional[List[str]]
+
+    @classmethod
+    def from_dict(cls, name: str, d: dict) -> 'Router':
+        return cls(
+            name=name,
+            url=d["name"],
+            type=d["type"],
+            network=d.get("network"),
+            routes=list(d["routes"]) if "routes" in d else None,
+        )
+
+@dataclasses.dataclass(frozen=True)
+class GrpcRoute:
+
+    @dataclasses.dataclass(frozen=True)
+    class MethodMatch:
+        type: Optional[str]
+        grpc_service: Optional[str]
+        grpc_method: Optional[str]
+        case_sensitive: Optional[bool]
+
+        @classmethod
+        def from_dict(cls, d: dict) -> 'MethodMatch':
+            return cls(
+                type=d.get("type"),
+                grpc_service=d.get("grpcService"),
+                grpc_method=d.get("grpcMethod"),
+                case_sensitive=d.get("caseSensitive"),
+            )
+
+    @dataclasses.dataclass(frozen=True)
+    class HeaderMatch:
+        type: Optional[str]
+        key: str
+        value: str
+
+        @classmethod
+        def from_dict(cls, d: dict) -> 'HeaderMatch':
+            return cls(
+                type=d.get("type"),
+                key=d["key"],
+                value=d["value"],
+            )
+
+    @dataclasses.dataclass(frozen=True)
+    class RouteMatch:
+        method: Optional['MethodMatch']
+        headers: Tuple['HeaderMatch']
+
+        @classmethod
+        def from_dict(cls, d: dict) -> 'RouteMatch':
+            return cls(
+                method=MethodMatch.from_dict(
```

Comment:
Hm, does this work for you? My IDE is complaining about unresolved reference for `MethodMatch`. And not as a incorrect type hint, as an actual error. Same for `Destination`, and `HeaderMatch`. I'll give your test a run with py3.6 to confirm.

Reply:
Confirming this works.

Reply:
Was definitely working on my end. I can confirm I've been using 3.6 in order to match the deployment version.